pkg/runtime/runtime.go

```diff
@@ -1096,33 +1098,36 @@ func (a *DaprRuntime) initNameResolution() error {
 		resolverVersion = components.FirstStableVersion
 	}
 
-	fName := utils.ComponentLogName(resolverName, "nameResolution", resolverVersion)
+	fName := utils.ComponentLogName("nr", resolverName, resolverVersion)
 	resolver, err = a.runtimeConfig.registry.NameResolutions().Create(resolverName, resolverVersion, fName)
+	if err != nil {
+		diag.DefaultMonitoring.ComponentInitFailed("nameResolution", "creation", resolverName)
+		return rterrors.NewInit(rterrors.CreateComponentFailure, fName, err)
+	}
+
 	resolverMetadata.Name = resolverName
 	if a.globalConfig.Spec.NameResolutionSpec != nil {
 		resolverMetadata.Configuration = a.globalConfig.Spec.NameResolutionSpec.Configuration
 	}
-	resolverMetadata.Properties = map[string]string{
-		nr.DaprHTTPPort: strconv.Itoa(a.runtimeConfig.httpPort),
-		nr.DaprPort:     strconv.Itoa(a.runtimeConfig.internalGRPCPort),
-		nr.AppPort:      strconv.Itoa(a.runtimeConfig.appConnectionConfig.Port),
-		nr.HostAddress:  a.hostAddress,
-		nr.AppID:        a.runtimeConfig.id,
+	resolverMetadata.Instance = nr.Instance{
```

Comment:
A small API change in contrib. Now properties such as address and port are passed in a strongly-typed struct rather than in a weakly-typed map.